### ISSUE TITLE
Window not resizing correctly

### ISSUE THREAD

**Issue by EternalClickbait** [stride3d/stride#1147]
**Release Type**: Official Release (4.1.0.1459-beta, .Net 5.0.8)

**Version**: ^^^

**Platform(s)**: Windows Desktop build

**Describe the bug**
When attempting to resize the game window (in windowed mode), the expected default behaviour of resizing does not occur, and the window will change it's size/position in ways it shouldn't (because something else was being changed)

**To Reproduce**
1. Build any project and run it (Debug/Release are the same)
2. Attempt to resize the game window that appears

**Expected behavior**
The window expands/contracts in the way that was desired. E.g. If dragging the right side of the window, it should increase in width, and have no other changes. 

**Actual Behaviour**
The game window does 'funky shit' that is completely wrong. Things I have experienced:
1. When attempting to make the window wider, the width stays the same, but the height shrinks instead (so the aspect ratio I was trying to get would probably be correct, but it changed the wrong thing)
2. Same thing for the height
3. The game window sets itself to 4k resolution, despite me being on a 1080p monitor.
4. The window will forcibly set it's width (and I believe height too sometimes) to a number when I attempt to resize that seems reasonably close to 4k resolution

**Screenshots**
https://cdn.discordapp.com/attachments/500292500641021954/887531412591357954/unknown.png
*It should be noted that I never made the window larger in any dimension (not purposefully), just tried to shrink it and move it. It enlarged itself on it's own*
Will also try to get recordings

**Log and callstacks**
N/A AFAIK

**Additional context**
When changing the resolution and or position via code, the window appears to act normally - the issue appears to be only through user interaction.

**Comment by EternalClickbait** [comment]
Here's my code that I used for testing. Most of it isn't important
*My Code*
```
using Stride.Core.Mathematics;
using Stride.Engine;
using Stride.Games;
using StrideSaber.EventManagement;
using StrideSaber.EventManagement.Events;
using System;
using System.Threading.Tasks;

namespace StrideSaber.Misc
{
	public class TestScript : AsyncScript
	{
		private static GameWindow Window;

		[EventMethod(typeof(GameStartedEvent))]
		private static void SetWindow(Event e)
		{
			Window = (e as GameStartedEvent)!.Game.Window;
		}

		/// <inheritdoc />
		public override async Task Execute()
		{
			int x = 960, y = 540;
			int flag = 0;
			int speed = 4;
			while (true)
			{
				await Task.Delay(10);
				if (Window is null)
				{
					Serilog.Log.Information("No window yet :(");
					continue;
				}
				switch (flag)
				{
					case 1:
						x += speed;
						break;
					case 2:
						y += speed;
						break;
					case 3:
						x -= speed;
						break;
					case 4:
						y -= speed;
						break;
					case 5:
						break;
					case 6:
						Window.SetSize(Window.PreferredWindowedSize);
						break;
				}

				if (x > 1280)
					x = 960;
				if (y > 720)
					y = 540;

				//If a console key was pressed, toggle our flag
				if (Console.KeyAvailable)
				{
					Console.ReadKey();
					flag++;
					if (flag > 6) flag = 1;
				}

				if ((flag != 6) && (flag != 5))
					Window.SetSize(new Int2(x, y));
				Serilog.Log.Information("Window: Bounds={WindowResolution}\tTarget={WindowTarget}\tFlag={Flag}", Window.ClientBounds.Size, Window.PreferredWindowedSize, flag);
			}
		}
	}
}
```

**Comment by MeharDT** [comment]
I'm not able to replicate this issue with a "New game" project in Stride 4.1.0.1459-beta using the provided instructions. Is Game.Window.AllowUserResizing set to true anywhere in your project? Does this issue also happen if AllowUserResizing is set to true in a blank "New game" project and you manually drag the window?

**Comment by phr00t** [comment]
@EternalClickbait do you have any Windows 10 display scaling enabled? I had to do a ton of changes to address that in my fork.

**Comment by EternalClickbait** [comment]
> 
> 
> @EternalClickbait do you have any Windows 10 display scaling enabled? I had to do a ton of changes to address that in my fork.

Nope, scaling is at 100%



> 
> 
> I'm not able to replicate this issue with a "New game" project in Stride 4.1.0.1459-beta using the provided instructions. Is Game.Window.AllowUserResizing set to true anywhere in your project? Does this issue also happen if AllowUserResizing is set to true in a blank "New game" project and you manually drag the window?

Not sure about new game, perhaps it's something special about my project. I am sure though that user resizing is enabled, because I have the code (and I know it's running). Also, If i set it to false, then no arrow appears to let me resize the windows (so it's impossible). Here's proof:

```csharp
using Serilog;
using SharpDX.Direct3D11;
using Stride.Core.Mathematics;
using Stride.Engine;
using Stride.Games;
using Stride.Graphics;
using StrideSaber.EventManagement;
using StrideSaber.EventManagement.Events;

namespace StrideSaber.Startup
{
	/// <summary>
	/// Handles events that occur when the game is started
	/// </summary>
	internal static class StartupEventHandler
	{
		[EventMethod(typeof(GameStartedEvent))]
		private static void SetWindowProperties(Event e)
		{
			Log.Debug("Enabling resizing of window");
			Game game = ((GameStartedEvent) e).Game;
			GameWindow window = game.Window;
			window.AllowUserResizing = true;

			Log.Debug("Setting size and position of window");
			window.Position = Int2.Zero;
			window.SetSize(new Int2(1280,720));
			Log.Error("{X}", window.ClientBounds);
		}
	}
}
```
![image](https://user-images.githubusercontent.com/48875125/133879554-99387be3-f57c-4b60-9cbc-bcc511bc5307.png)

**Update:**
Tested with both SDL and WPF (`game.Run(GameContextFactory.NewGameContextWpf());`//`game.Run(GameContextFactory.NewGameContextSDL());`) and the issue is the same. I'm sure this is a code error not a problem with compatability

**Comment by EternalClickbait** [comment]
**Project Link**
@MeharDT Try downloading the project and running it, then tell me if you experience the same issue
https://github.com/EternalClickbait/StrideSaber/tree/4dffab1b664e6d525a55638001913e5db3362a40

**Comment by Eideren** [comment]
Set ``AdaptBackBufferToScreen`` to false in your GameSettings, looks like it's kind of broken right now.
It should be fairly straight forward if anyone wants to tackle this:
https://github.com/stride3d/stride/blob/master/sources/engine/Stride.Engine/Engine/Game.cs#L317
https://github.com/stride3d/stride/blob/master/sources/engine/Stride.Games/GameBase.cs#L425

**Comment by EternalClickbait** [comment]
> 
> 
> Set `AdaptBackBufferToScreen` to false in your GameSettings, looks like it's kind of broken right now.
> It should be fairly straight forward if anyone wants to tackle this:
> https://github.com/stride3d/stride/blob/master/sources/engine/Stride.Engine/Engine/Game.cs#L317
> https://github.com/stride3d/stride/blob/master/sources/engine/Stride.Games/GameBase.cs#L425
> And we would want to find a better name for that property as this one misrepresent how it is used/described

Yeah that seems to fix it. Do you know why it seems to be doing this?

**Comment by Eideren** [comment]
``PreferredBackBufferWidth``, ``RequestedWidth`` and ``Window.ClientBounds``  are not properly synchronized, looking at/debugging the lines I pointed to should give you a better understanding.
